include/rocksdb/customizable.h

```diff
@@ -87,7 +87,18 @@ class Customizable : public Configurable {
   // @param name The name of the instance to find.
   // Returns true if the class is an instance of the input name.
   virtual bool IsInstanceOf(const std::string& name) const {
-    return name == Name();
+    if (name.empty()) {
+      return false;
+    } else if (name == Name()) {
+      return true;
+    } else {
+      auto sname = ShortName();
```

Comment:
Can we avoid using auto here, it is better to give clear type in the public portion.